### ISSUE TITLE
Inconsistent doc and code: matmul bf16 reduction setting

### ISSUE THREAD

**Issue by lkct** [pytorch/pytorch#100966]
### 🐛 Describe the bug

In the [docs on reduced precision bf16 gemm](https://pytorch.org/docs/main/notes/cuda.html#reduced-precision-reduction-in-bf16-gemms), it is stated that:
> Note that this switch is set to *False* by default for BF16 as we have observed numerical instability in PyTorch CI tests (e.g., test/test_matmul_cuda.py).

However the actual behaviour is *True*:
```python
import torch
print(torch.backends.cuda.matmul.allow_bf16_reduced_precision_reduction)  # True
```

### Versions

PyTorch version: 2.0.0+cu117
Is debug build: False
CUDA used to build PyTorch: 11.7
ROCM used to build PyTorch: N/A

OS: Ubuntu 20.04.6 LTS (x86_64)
GCC version: (Ubuntu 9.4.0-1ubuntu1~20.04.1) 9.4.0
Clang version: Could not collect
CMake version: version 3.26.3
Libc version: glibc-2.31

Python version: 3.8.16 (default, Mar  2 2023, 03:21:46)  [GCC 11.2.0] (64-bit runtime)
Python platform: Linux-5.15.0-52-generic-x86_64-with-glibc2.17
Is CUDA available: True
CUDA runtime version: 11.7.99
CUDA_MODULE_LOADING set to: LAZY

cc @svekars @carljparker

**Comment by ezyang** [comment]
@eqy could you please take a look

**Comment by lkct** [comment]
(should be closed by the above PR)

**Comment by lkct** [comment]
Sorry I missed it before. Just found the same issue in another place: docs for [Numerical accuracy](https://pytorch.org/docs/main/notes/numerical_accuracy.html#reduced-precision-reduction-for-fp16-and-bf16-gemms)
> A similar flag exists for BF16 GEMM operations and is turned off by default.

Not sure if there are any more references to this in the docs.
@ezyang @eqy could you please check it out? Thanks
